AWS Appstream 2.0 crashes

### Terraform CLI and Terraform AWS Provider Version

Terraform v1.0.2
on windows_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v4.16.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.11.0
+ provider registry.terraform.io/hashicorp/local v1.4.0
+ provider registry.terraform.io/hashicorp/random v3.3.0
+ provider registry.terraform.io/hashicorp/tls v3.4.0

Your version of Terraform is out of date! The latest version
is 1.2.2. You can update by downloading from https://www.terraform.io/downloads.html


### Affected Resource(s)

AWS Appstream 2.0

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_appstream_fleet" "example" {
  name = "AppFleet_${var.env}"
  compute_capacity {
    desired_instances = 1
  }
  description                        = "${var.env} fleet"
  idle_disconnect_timeout_in_seconds = 60
  display_name                       = "${var.env}-fleet"
  enable_default_internet_access     = true
  fleet_type                         = "ON_DEMAND"
  image_name                         = var.images.streaming
  instance_type                      = "stream.standard.small"
  max_user_duration_in_seconds       = 900
  iam_role_arn = aws_iam_role.appstream-agent-role.arn
  vpc_config {
    subnet_ids = [data.aws_subnet.app_public_subnet_az1.id]
  }
  tags = var.tags
}

resource "aws_appstream_stack" "example" {
  name         = "AppStack_${var.env}"
  description  = "App ${var.env} stack"
  display_name = "App_${var.env}"


  application_settings {
    enabled        = true
    settings_group = "SettingsGroup"
  }

  tags = var.tags
}


resource "aws_appstream_fleet_stack_association" "stack_fleet_association" {
  fleet_name = aws_appstream_fleet.example.name
  stack_name = aws_appstream_stack.example.name
}

```

### Debug Output


### Panic Output

Stack trace from the terraform-provider-aws_v4.16.0_x5.exe plugin:

panic: interface conversion: interface {} is []interface {}, not *schema.Set

goroutine 97 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/appstream.resourceStackUpdate({0xadd20c0, 0xc002b81dd0}, 0xc002b45c80, {0x8858e00, 0xc000ef4a80})
        github.com/hashicorp/terraform-provider-aws/internal/service/appstream/stack.go:338 +0xbe5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xadd20c0, {0xadd20c0, 0xc002b81dd0}, 0xd, {0x8858e00, 0xc000ef4a80})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:736 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000a20a80, {0xadd20c0, 0xc002b81dd0}, 0xc0018355f0, 0xc002b45b00, {0x8858e00, 0xc000ef4a80})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/resource.go:847 +0x9e5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000290d38, {0xadd2018, 0xc000a745c0}, 0xc002af7450)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0019ce960, {0xadd20c0, 0xc002b815f0}, 0xc002b8c000)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:812 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x9a60020, 0xc0019ce960}, {0xadd20c0, 0xc002b815f0}, 0xc002b46420, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000228c40, {0xaedf010, 0xc000df04e0}, 0xc0029230e0, 0xc0019d2a50, 0x10b23c60, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000228c40, {0xaedf010, 0xc000df04e0}, 0xc0029230e0, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x294

Error: The terraform-provider-aws_v4.16.0_x5.exe plugin crashed!


### Expected Behavior

apply the stack association between to the existing fleet

### Actual Behavior

Crashes

### Steps to Reproduce

1. create an appstream definition
2. `terraform apply`

I think the issue is around:

aws_appstream_stack with the property `embed_host_domains`.
When set to : 

```hcl
resource "aws_appstream_stack" "passpro_next_test_stack" {


...


  embed_host_domains = ["app-${var.env}.somedomain.com"]

.... 

}

```


Stack trace from the terraform-provider-aws_v4.18.0_x5 plugin:

panic: interface conversion: interface {} is *schema.Set, not []interface {}

goroutine 109 [running]:
github.com/hashicorp/terraform-provider-aws/internal/service/appstream.resourceStackCreate({0xa65fb20, 0xc0016c7800}, 0xc002242000, {0x80b53c0, 0xc000221500})
        github.com/hashicorp/terraform-provider-aws/internal/service/appstream/stack.go:214 +0xa76
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xa65fb20, {0xa65fb20, 0xc0016c7800}, 0xd, {0x80b53c0, 0xc000221500})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/schema/resource.go:702 +0x84
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000bc2c40, {0xa65fb20, 0xc0016c7800}, 0xc002af5040, 0xc001b67e80, {0x80b53c0, 0xc000221500})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/schema/resource.go:837 +0xc29
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000285128, {0xa65fa78, 0xc0023754c0}, 0xc0016d0780)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.17.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001b36960, {0xa65fb20, 0xc0016c7020}, 0xc0016ac620)
        github.com/hashicorp/terraform-plugin-go@v0.9.1/tfprotov5/tf5server/server.go:812 +0x56b
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x92d2d60, 0xc001b36960}, {0xa65fb20, 0xc0016c7020}, 0xc002214240, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.1/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000242700, {0xa76e0f0, 0xc00101c1a0}, 0xc0021db9e0, 0xc001b33170, 0x1042c260, 0x0)
        google.golang.org/grpc@v1.46.0/server.go:1283 +0xcf2
google.golang.org/grpc.(*Server).handleStream(0xc000242700, {0xa76e0f0, 0xc00101c1a0}, 0xc0021db9e0, 0x0)
        google.golang.org/grpc@v1.46.0/server.go:1620 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.46.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.46.0/server.go:920 +0x294

Error: The terraform-provider-aws_v4.18.0_x5 plugin crashed!

--- a/.changelog/25372.txt
+++ b/.changelog/25372.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+data-source/aws_appstream_stack: Fix crash when setting `embed_host_domains`
+```

--- a/.changelog/25372.txt
+++ b/.changelog/25372.txt
@@ -1,3 +1,3 @@
 ```release-note:bug
-data-source/aws_appstream_stack: Fix crash when setting `embed_host_domains`
+resource/aws_appstream_stack: Fix crash when setting `embed_host_domains`
 ```

--- a/internal/service/appstream/stack.go
+++ b/internal/service/appstream/stack.go
@@ -211,7 +211,7 @@ func resourceStackCreate(ctx context.Context, d *schema.ResourceData, meta inter
 	}
 
 	if v, ok := d.GetOk("embed_host_domains"); ok {
-		input.EmbedHostDomains = flex.ExpandStringList(v.([]interface{}))
+		input.EmbedHostDomains = flex.ExpandStringList(v.(*schema.Set).List())
 	}
 
 	if v, ok := d.GetOk("feedback_url"); ok {

--- a/internal/service/appstream/stack_test.go
+++ b/internal/service/appstream/stack_test.go
@@ -95,6 +95,9 @@ func TestAccAppStreamStack_complete(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "name", rName),
 					acctest.CheckResourceAttrRFC3339(resourceName, "created_time"),
 					resource.TestCheckResourceAttr(resourceName, "description", descriptionUpdated),
+					resource.TestCheckResourceAttr(resourceName, "embed_host_domains.#", "2"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "embed_host_domains.*", "example.com"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "embed_host_domains.*", "subdomain.example.com"),
 				),
 			},
 			{
@@ -214,6 +217,8 @@ func testAccStackCompleteConfig(name, description string) string {
 resource "aws_appstream_stack" "test" {
   name        = %[1]q
   description = %[2]q
+
+  embed_host_domains = ["example.com", "subdomain.example.com"]
 
   storage_connectors {
     connector_type = "HOMEFOLDERS"

